students/tfbanks/Lesson04/Mailroom_pt2.py

```diff
@@ -0,0 +1,87 @@
+    letter = ('\n\nDear {},\n'
+              '\nThank you for your generous donation of ${:,.2f}, your generosity is greatly appreciated.\n'
+              'These funds will help continue our efforts to teach Python to the next generation.\n'
+              '\nSincerely,\n'
+              '\nJoesef Edword Bringingham\n\n')
+    if ty_donor == 'List':  # If List, Prints Donor List and restarts the function
+        for donor in donor_names.keys():
+            print(donor)
+        thank_you()
+    elif ty_donor in donor_names:  # Actions if donor is already in donor_names
+        amount = float(input('Please enter the donation amount: '))
+        donor_names[ty_donor].append(amount)  # Appends donation amount to donators donation values
+        print(letter.format(ty_donor, amount))
+        what_to_do()  # Takes user back to action options
+    elif ty_donor not in donor_names:  # If donor is not on the list, this collects the data, adds to the donor_names dictionary, and writes a new letter.
+        amount = float(input('Please enter the donation amount: '))
+        donor_names[ty_donor] = [amount]
+        print(letter.format(ty_donor, amount))
+        what_to_do()  # Takes user back to action options
+
+
+def report():  # Defines parameters of the report of donors and prints it.
+    print('\nDonor Name            |  Total Given | Num Gifts |  Average Gift')
+    print('------------------------------------------------------------------')
+    for donor, donations in donor_names.items():
+        print('{:24s}'.format(donor), '${:11,.2f}'.format(sum(donations)), '{:^12}'.format(len(donations)),
+              '${:12,.2f}'.format(sum(donations)/len(donations)))
+    print('\n')
+    what_to_do()  # takes the user to another what to do statement that reflects they just ran a report
+
+
+def mass_mail():
+
+    mm_letter = ('\n\nDear {},\n'
+              '\nThank you for your generous donations totaling ${:,.2f}, your generosity is greatly appreciated.\n'
+              'These funds will help continue our efforts to teach Python to the next generation.\n'
+              '\nSincerely,\n'
+              '\nJoesef Edword Bringingham\n\n')
+    for donor, donations in donor_names.items():
+        letter = mm_letter.format(donor, sum(donations))
+        letter_output = donor + '.txt'
+        with open(letter_output, 'w') as op:
+            for line in letter:
+                op.write(line)
+    print('\n\nLetters are located in your working directory folder\n\n')
+    what_to_do()
+
+
+def quit_option():
+    print('Have a nice day')
+    quit()
+
+
+def what_to_do():
+    answer = input('Please choose an action from the following options:\n\n'
+                   '1 - Send a Thank You\n'
+                   '2 - Create a Report\n'
+                   '3 - Send letters to everyone\n'
+                   '4 - Quit\n')
+    while answer not in ['1', '2', '3', '4']:
+        what_to_do()
+    selection(answer)()
```

Comment:
how about this:
```
while True:
    answer = input("")
    select(answer)  # might as well call the function in here too, otherwise `selection` func not providing much value
```
this is it, that's all that should be in your `what_to_do` function; and remove calls to `what_to_do` in other functions.